website/docs/cli/commands/test.mdx

```diff
@@ -1,12 +1,165 @@
+├── README.md
+├── main.tf
+├── main.tftest.hcl
+├── foo.tf
+├── foo.tftest.hcl
+├── bar.tf
+├── bar.tftest.hcl
+├── outputs.tf
+└── variables.tf
+```
+
+**WHEN** `tofu test` is executed in the project directory,
+
+**THEN** the tests defined in all three files, `main.tftest.hcl`, `foo.tftest.hcl` and `bar.tftest.hcl`, will be run.
+
+**WHEN** `tofu test -filter=foo.tftest.hcl -filter=bar.tftest.hcl` is executed in the project directory,
+
+**THEN** the tests defined in the files `foo.tftest.hcl` and `bar.tftest.hcl` will be run only.
+
+### Nested structure
+
+**GIVEN** the following OpenTofu module structure:
+
+```commandline
+.
+├── LICENSE
+├── README.md
+├── foo.tf
+├── foo.tftest.hcl
+├── bar.tf
+├── bar.tftest.hcl
+├── main.tf
+├── main.tftest.hcl
+├── tests
+│   ├── foo.tftest.hcl
+│   └── bar.tftest.hcl
+├── modules
+│   ├── module_foo
+│   │   ├── LICENSE
+│   │   ├── README.md
+│   │   ├── foo.tf
+│   │   ├── foo.tftest.hcl
+│   │   ├── main.tf
+│   │   ├── main.tftest.hcl
+│   │   ├── outputs.tf
+│   │   └── variables.tf
+│   └── module_bar
+│       ├── LICENSE
+│       ├── README.md
+│       ├── foo.tf
+│       ├── foo.tftest.hcl
+│       ├── bar.tf
+│       ├── bar.tftest.hcl
+│       ├── main.tf
+│       ├── main.tftest.hcl
+│       ├── outputs.tf
+│       └── variables.tf
+├── outputs.tf
+└── variables.tf
+```
+
+**WHEN** `tofu test` is executed in the project root directory,
+
+**THEN** the tests defined in the following files will be run (note that the files are sorted in the execution order):
+
+- bar.tftest.hcl,
+- foo.tftest.hcl,
+- main.tftest.hcl,
+- tests/bar.tftest.hcl,
+- tests/foo.tftest.hcl.
+
+**WHEN** `tofu test -test-directory=modules/module_bar` is executed in the project root directory,
+
+**THEN** the tests defined in the following files will be run (note that the files are sorted in the execution order):
+
+- bar.tftest.hcl,
+- foo.tftest.hcl,
+- main.tftest.hcl,
+- modules/module_bar/bar.tftest.hcl,
+- modules/module_bar/foo.tftest.hcl,
+- modules/module_bar/main.tftest.hcl.
+
+**WHEN** `tofu test -test-directory=modules/module_bar -filter=bar.tftest.hcl` is executed in the project root directory,
+
+**THEN** the tests defined in the file `bar.tftest.hcl` will be run only.
+
+**WHEN** `tofu test -test-directory=modules/module_bar -filter=modules/module_bar/bar.tftest.hcl` is executed in the project root directory,
+
+**THEN** the tests defined in the file `modules/module_bar/bar.tftest.hcl` will be run only.
+
+## Definition syntax
+
+TBD
+
+## Variables
+
+Tests can be defined using variables specified in several ways.
+
+The variable values will be resolved according to the following order:
+
+1. Environment variables with the prefix `TF_VAR_`;
+2. tfvar files specified in terraform.tfvars, *.auto.tfvars;
+3. Files specified using the flag `-var-file`;
+4. Variables defined using the flag `-var`;
+5. Variables from the `variable block` in test file;
+6. Variables from the variable block in run block.
```

Comment:
I think that we should already have docs on variable precedence. Maybe we can link there instead

Reply:
Could you please confirm if you mean this [subsection](http://localhost:3000/docs/language/values/variables/#variable-definition-precedence)? Thanks!